### PR TITLE
[MODULAR] Medicell Safety

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
Gives the Medicells an option to toggle the safety on them by using them in hand. Toggling the safety means that they are unable to heal damage if it will cause any kind of clone damage to the patient.  This should make it that healing someone at high damage thresholds with medigun will be much more of an intentional choice. By default, cells have safety on.

## Why It's Good For The Game
Prevents doctors from accidentally doing clone damage when using the brute/burn cells on the medigun. 

## Changelog
:cl:
expansion: Medicells now have a safety toggle, preventing them from healing brute/burn when it would do clone damage.
refactor: refactors a bunch of code pertaining to the healing on medicells. 
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm [hunk 1]
```diff
@@ -71,6 +71,27 @@
...
+//MEDIGUN SAFETY
+/obj/item/medicell/attack_self(mob/living/user)
+	if(!has_safety)
+		return
+	if(!on_safety)
+		on_safety = TRUE
+		to_chat(user, span_notice("The safety on the Medicell is now on, you cannot heal when it will cause clone damage to the patient"))
+		src.ammo_type = safe_ammo
+		return
+	if(on_safety ==  1)
```
Comment (tf-4) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm]:
```suggestion
	if(on_safety)
```
Tiny tweak.

Comment (softcerv) [reply modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm]: ah, thank you!

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 2]
```diff
@@ -115,6 +115,44 @@
...
+	icon_state = "red_laser"
+
+/obj/projectile/energy/medical/safe/brute1/on_hit(mob/living/target)
+	. = ..()
+	if(!istype(target, /mob/living/carbon/human))
+		return
+	if(target.stat == DEAD)
+		return
+	//Stops healing from 50 or over
+	if(target.getBruteLoss() > 49)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]: Logically the same thing, but it just looks better code-wise.
```suggestion
	if(target.getBruteLoss() >= 50)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 3]
```diff
@@ -115,6 +115,44 @@
...
+	icon_state = "yellow_laser"
+
+/obj/projectile/energy/medical/safe/burn1/on_hit(mob/living/target)
+	. = ..()
+	if(!istype(target, /mob/living/carbon/human))
+		return
+	if(target.stat == DEAD)
+		return
+	//Stops healing from 50 or over
+	if(target.getFireLoss() > 49)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]: Ditto.
```suggestion
	if(target.getFireLoss() >= 50)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 4]
```diff
@@ -132,9 +170,9 @@
 		return
 	if(target.stat == DEAD)
 		return
-	//DISGUST 
+	//DISGUST
 	if(target.getBruteLoss() > 49)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]: Ditto.
```suggestion
	if(target.getBruteLoss() >= 50)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 5]
```diff
@@ -132,9 +170,9 @@
...
-		target.adjust_disgust(1.5) 
+		target.adjust_disgust(1.5)
 	if(target.getBruteLoss() > 99)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]: Ditto.
```suggestion
	if(target.getBruteLoss() >= 100)
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
There's a _lot_ of repetition going on here.
Try making something that's less repetitive, this could be a very good code improvement if you reduced all this repetition in here. If you need help with that, hit me up on Discord and we can go through it together.
However, this isn't ready for merging yet, there's simply too many things that repeat themselves, and there's some occasions where you make several identical proc calls in a row, like for `target.adjust_disgust()`, which would gain from using a temporary variable to store by how much you want the disgust to be modified, and only be called once if the disgust delta is different from 0, for instance.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I think you should try having the medicells have three subtypes, `brute`, `burn`, `toxin` and then you just tweak the values this way, it would be much cleaner, I'd imagine.

**Comment by Gandalf2k15** [comment]
Is this done?@

**Review by GoldenAlpharex — APPROVED** [review]
I'm satisfied with the code quality, if this works as intended still, it's good to merge.

**Comment by softcerv** [comment]
> 
> 
> I'm satisfied with the code quality, if this works as intended still, it's good to merge.

I did testing on everything, it should probably work.

**Pull request merged**